contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -209,39 +212,63 @@ func (t *T) Reset() {
 	t.Spec = Spec{}
 }
 
+func (t *T) computeResourceLimit(name string, defaultLimit float64) float64 {
+	var x float64 = 0.0
+	for _, container := range t.Pod.Spec.Containers {
+		for limitName, limit := range container.Resources.Limits {
+			if string(limitName) == name {
+				value := limit.Value()
+				if value > 1000000 { // avoid overflow with MilliValue
+					x += float64(value)
+				} else {
+					x += float64(limit.MilliValue()) / 1000.0
+				}
+			}
+		}
+	}
+	if x == 0.0 {
```

Comment:
Reusing the resourcequota functions now to extract limits. Otherwise, I have unified all the scheduler resource logic to use MB, i.e. the mesos memory unit. This simplifies the file a lot IMO.